src/vanilla/Model/CodeModelTS.cs

```diff
@@ -92,12 +92,18 @@ public string BasePath
             }
         }
 
-        /// <summary>
-        /// Get whether or not this CodeModelTS has any model types.
-        /// </summary>
-        public bool HasModelTypes()
+        private IEnumerable<string> allModelTypeNames;
+        [JsonIgnore]
+        public IEnumerable<string> AllModelTypeNames
         {
-            return ModelTypes.Any();
+            get
+            {
+                if (allModelTypeNames == null)
+                {
+                    allModelTypeNames = AllModelTypes.Select((CompositeType modelType) => modelType.Name.ToString()).ToHashSet();
```

Comment:
Why HashSet?

Reply:
Risk of dupes?

Reply:
Not because of possible duplicates. I went with a set because I was hoping that would be the best for performance. It should be a constant-time lookup compared to a sorted array's logarithmic time and a non-sorted array's linear time.

Reply:
Oh, because the .Contains will direct to the HashSet implementation. Got it.